wdl-analysis/src/queue.rs

```diff
@@ -654,51 +711,59 @@ where
         }
 
         let total = tasks.len();
-        self.tokio
-            .block_on((self.progress)(context.clone(), kind, 0, total));
+        if completed.is_some() {
```

Comment:
Is this right? Seems like we shoudl send progress updates even if the completed callback is not provided. Perhaps I'm missing something with this.

Reply:
This was to make the analysis **block** in case on completions and **cancellable** in other cases - what i asked in slack that we would need analyzed document to access methods like `structs`, `namespaces`, `tasks` etc on it